pkg/gadgets/run/tracer/tracer.go

```diff
@@ -1052,6 +1081,30 @@ func (t *Tracer) runToppers(gadgetCtx gadgets.GadgetContext) {
 	}
 }
 
+func (t *Tracer) collectProfiler(gadgetCtx gadgets.GadgetContext) error {
+	cb := t.processEventFunc(gadgetCtx)
+
+	events := []*types.Event{}
+
+	key := make([]byte, t.histMap.KeySize(), t.histMap.KeySize())
+	value := make([]byte, t.histMap.ValueSize(), t.histMap.ValueSize())
```

Comment:
You are using a great generic approach to iterate over the map, so why do we need the developer to provide the keyType in the `GADGET_PROFILER` macro?